presto-main/src/main/java/com/facebook/presto/memory/RemoteNodeMemory.java

```diff
@@ -137,4 +157,24 @@ public void onFailure(Throwable t)
             }, directExecutor());
         }
     }
+
+    private StaticBodyGenerator createBodyGenerator(MemoryPoolAssignmentsRequest assignments)
+    {
+        if (isBinaryTransportEnabled) {
+            return smileBodyGenerator((SmileCodec<MemoryPoolAssignmentsRequest>) assignmentsRequestCodec, assignments);
+        }
+        return jsonBodyGenerator(unwrapJsonCodec(assignmentsRequestCodec), assignments);
+    }
+
+    private Builder setBinaryTransportHeadersIfNeeded(Builder requestBuilder)
```

Comment:
We should think of how can this be a part of the `BodyGenerator`. Maybe we should introduce a slightly different interface, that can both, encode the response and set the headers.